### PR TITLE
Print value fixes v2, fixes roottest

### PR DESCRIPTION
**Pull request by perovic**
**Removed "VALID" after address output, fixes roottest.**
Various fixes based on comments after second round.

*More changes to come (char removal, CATCHALL, potential real types format refinements).*

### CODE REVIEW

File: interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h [hunk 1]
```diff
@@ -59,6 +58,7 @@ namespace cling {
   // std::string
   std::string printValue(const std::string &val);
 
+  class Value;
```
Comment (karies) [inline interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h]: Can you move this up, right after `namespace cling {`?

File: interpreter/cling/lib/Interpreter/ValuePrinter.cpp [hunk 2]
```diff
@@ -348,7 +348,7 @@ static std::string printUnpackedClingValue(const Value& V) {
 
   if(Ty-> isNullPtrType()) {
     // special case nullptr_t
-    strm << "<<<NULL>>>";
+    strm << "nullptr";
```
Comment (karies) [inline interpreter/cling/lib/Interpreter/ValuePrinter.cpp]: That should be `nullptr_t` (the type, not the value).

File: interpreter/cling/lib/Interpreter/ValuePrinter.cpp [hunk 3]
```diff
@@ -369,42 +369,52 @@ static std::string printUnpackedClingValue(const Value& V) {
...
-    std::ostringstream strm;
-    strm << std::boolalpha;
-    strm << val;
-    strm << std::noboolalpha;
-    return strm.str();
+    if (val) {
+      return "true";
+    } else {
+      return "false";
+    }
```
Comment (karies) [inline interpreter/cling/lib/Interpreter/ValuePrinter.cpp]: Alternatively, `return val ? "true" : "false"` would have done :-) But details, whatever you prefer.

File: interpreter/cling/lib/Interpreter/ValuePrinter.cpp [hunk 4]
```diff
@@ -458,13 +468,13 @@ namespace cling {
 
   std::string printValue(float val) {
     std::ostringstream strm;
-    strm << val;
+    strm << std::showpoint << val << "f";
```
Comment (karies) [inline interpreter/cling/lib/Interpreter/ValuePrinter.cpp]: Awesome!

File: interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h (hunk 1 above)
Comment (perovic) [inline interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h]: ✓

File: interpreter/cling/lib/Interpreter/ValuePrinter.cpp (hunk 2 above)
Comment (perovic) [inline interpreter/cling/lib/Interpreter/ValuePrinter.cpp]: Yes, this is interesting.

File: interpreter/cling/lib/Interpreter/ValuePrinter.cpp (hunk 3 above)
Comment (perovic) [inline interpreter/cling/lib/Interpreter/ValuePrinter.cpp]: Ternary operation wins.

File: interpreter/cling/test/Prompt/ValuePrinter/Assignments.C [hunk 5]
```diff
@@ -47,7 +47,7 @@ e2
 
 
 // Arrays:
-float farr[] = {0.,1.,2.,3.,4.,5.} // CHECK: (float [6]) { 0, 1, 2, 3, 4, 5 }
+float farr[] = {0.,1.,2.,3.,4.,5.} // CHECK: (float [6]) { 0.00000f, 1.00000f, 2.00000f, 3.00000f, 4.00000f, 5.00000f }
```
Comment (karies) [inline interpreter/cling/test/Prompt/ValuePrinter/Assignments.C]: Cool. Congrats!

File: interpreter/cling/lib/Interpreter/ValuePrinter.cpp (hunk 4 above)
Comment (perovic) [inline interpreter/cling/lib/Interpreter/ValuePrinter.cpp]: :+1:

File: interpreter/cling/test/Prompt/ValuePrinter/Regression.C [hunk 6]
```diff
@@ -47,7 +47,7 @@ q // CHECK: (const int *) 0x123
 // PR ROOT-5467
 &A::someFunc // CHECK: (int (A::*)(float)) Function @0x{{[0-9a-f]+}}
 
-nullptr // CHECK: (nullptr_t) <<<NULL>>>
+nullptr // CHECK: (nullptr_t) nullptr
```
Comment (karies) [inline interpreter/cling/test/Prompt/ValuePrinter/Regression.C]: Sweet. C++11!

File: interpreter/cling/test/Prompt/ValuePrinter/Assignments.C (hunk 5 above)
Comment (perovic) [inline interpreter/cling/test/Prompt/ValuePrinter/Assignments.C]: We can refine this further later, maybe to a different number of decimals.

**Comment by karies** [comment]
Thanks, merged!

**Pull request merged**